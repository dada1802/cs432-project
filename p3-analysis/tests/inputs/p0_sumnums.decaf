int nums[100];

def int sum_nums1(int len)
{
    int count;
    int sum;
    count = 0;
    sum = 0;
    while (count < len) {
        sum = sum + nums[count];
        count = count + 1;
    }
    return sum;
}

def int sum_nums2(int len)
{
    int sum;
    int i;
    sum = 0;
    i = 0;
    while (i < len)
    {
        sum = sum + nums[i];
        i = i + 1;
    }
    return sum;
}

def int sum_nums3(int len)
{
    int i;
    int sum;
     nums[0] = 1;
     nums[1] = 2;
     nums[2] = 3;
     nums[3] = 4;
     nums[4] = 5;
     nums[5] = 6;
     nums[6] = 7;
     nums[7] = 8;
     i = 0;
     sum = 0;
     // return 0 if 0
    if(len == 0)
    {
        return 0;
    }

    while(i < len && i < 100)
    {
        sum = sum + nums[i];
         i = i + 1;
    }
    return sum;
}

def int main()
{
    return 0;
}