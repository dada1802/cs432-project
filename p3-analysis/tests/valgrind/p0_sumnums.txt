==135096== Memcheck, a memory error detector
==135096== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==135096== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==135096== Command: ../decaf inputs/p0_sumnums.decaf
==135096== 
Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 sum_nums1 : (int) -> int
 sum_nums2 : (int) -> int
 sum_nums3 : (int) -> int
 main : () -> int
 nums : int [100]

  FuncDecl name="sum_nums1" return_type=int parameters={len:int} [line 3]
  SYM TABLE:
   len : int

    Block [line 4]
    SYM TABLE:
     count : int
     sum : int

        Block [line 9]
        SYM TABLE:

  FuncDecl name="sum_nums2" return_type=int parameters={len:int} [line 16]
  SYM TABLE:
   len : int

    Block [line 17]
    SYM TABLE:
     sum : int
     i : int

        Block [line 23]
        SYM TABLE:

  FuncDecl name="sum_nums3" return_type=int parameters={len:int} [line 30]
  SYM TABLE:
   len : int

    Block [line 31]
    SYM TABLE:
     i : int
     sum : int

        Block [line 46]
        SYM TABLE:

        Block [line 51]
        SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 58]
  SYM TABLE:

    Block [line 59]
    SYM TABLE:

==135096== 
==135096== HEAP SUMMARY:
==135096==     in use at exit: 0 bytes in 0 blocks
==135096==   total heap usage: 2,879 allocs, 2,879 frees, 375,520 bytes allocated
==135096== 
==135096== All heap blocks were freed -- no leaks are possible
==135096== 
==135096== For lists of detected and suppressed errors, rerun with: -s
==135096== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
