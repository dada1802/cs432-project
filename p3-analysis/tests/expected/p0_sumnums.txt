Program [line 1]
SYM TABLE:
 print_int : (int) -> void
 print_bool : (bool) -> void
 print_str : (str) -> void
 sum_nums1 : (int) -> int
 sum_nums2 : (int) -> int
 sum_nums3 : (int) -> int
 main : () -> int
 nums : int [100]

  FuncDecl name="sum_nums1" return_type=int parameters={len:int} [line 3]
  SYM TABLE:
   len : int

    Block [line 4]
    SYM TABLE:
     count : int
     sum : int

        Block [line 9]
        SYM TABLE:

  FuncDecl name="sum_nums2" return_type=int parameters={len:int} [line 16]
  SYM TABLE:
   len : int

    Block [line 17]
    SYM TABLE:
     sum : int
     i : int

        Block [line 23]
        SYM TABLE:

  FuncDecl name="sum_nums3" return_type=int parameters={len:int} [line 30]
  SYM TABLE:
   len : int

    Block [line 31]
    SYM TABLE:
     i : int
     sum : int

        Block [line 46]
        SYM TABLE:

        Block [line 51]
        SYM TABLE:

  FuncDecl name="main" return_type=int parameters={} [line 58]
  SYM TABLE:

    Block [line 59]
    SYM TABLE:

